File: src/link.rs
// Symlink / junction creation.
//
// The actual platform-specific logic lives in `os::create_link`. This module
// is a thin wrapper that returns i32 (mirroring rosie_create_link in
// link.c) and emits a log_error on failure.

use std::path::Path;

/// Create a directory-or-file link from `link_path` to `target`. Returns 0
/// on success, -1 on failure (with a log::error already emitted).
pub fn rosie_create_link(target: &Path, link_path: &Path, is_dir: bool) -> i32 {
    match crate::os::create_link(target, link_path, is_dir) {
        Ok(()) => 0,
        Err(e) => {
            crate::log::error(&format!("{e}"));
            -1
        }
    }
}

File: npm/rosie-skills/src/link.ts
// Symlink / junction creation. Ported from src/link.rs.
//
// The platform-specific logic lives in os.createLink. This is a thin wrapper
// that returns a number (0 ok, -1 fail) and emits a log.error on failure.

import * as os from "./os.js";
import * as log from "./log.js";

// Create a directory-or-file link from linkPath to target. Returns 0 on
// success, -1 on failure (with a log.error already emitted).
export function rosieCreateLink(target: string, linkPath: string, isDir: boolean): number {
  try {
    os.createLink(target, linkPath, isDir);
    return 0;
  } catch (e) {
    log.error(e instanceof Error ? e.message : String(e));
    return -1;
  }
}
